### DISCUSSION TITLE
Adding attributes other than ID to a group tag

### DISCUSSION THREAD

**Discussion by huynhj02** [jfree/jfreesvg#42] [Q&A] [answered]
Hi! I am using JFreeSVG for a project and was wondering if there was any way I could add additional attributes onto a group tag? 

Ex: currently I know I can add an id to the group tag, but if I wanted to add an attibute like "filepath" to my group tag (so that the tag would read something like < g id="unique-id" filepath="unique-filepath" >, is there an easy way to do so? 

Thank you so much for your time!

EDIT: Oops, I didn't realize the example wasn't visible, added spaces between the < > brackets.

**Comment by jfree** [comment, marked as answer]
That's not supported at present but I suppose the rendering hint `SVGHints.KEY_BEGIN_GROUP` could be modified to support it.  Right now it expects a `String` instance for the value and uses that as the group ID.  It could check for an instance of `Map<String, String>` and consider each entry in the map to be an attribute-value pair and simply write those into the SVG output.  Would that work for you?

**Reply by huynhj02** [reply]
Yeah! That solution would be great if it could be supported.

**Reply by huynhj02** [reply]
Hey! I was planning on making a PR to add this feature in, but it turns out it was already supported! The `SVGHints.KEY_BEGIN_GROUP` can take a value of `Map<String,Object>`, which is then used to create an attribute-value pair. See Line 979 of SVGGraphics2D.java for more, just wanted to leave a note in case anyone else wanted to use such a feature.

**Reply by jfree** [reply]
So, first of all I'm glad the feature is there already!  But then, I apologise for not remembering it.  In fact I was mislead by the Javadocs, so I'm about to commit an update for that.

**Reply by huynhj02** [reply]
No worries! Thank you for maintaining and updating this library, really appreciate it.
